### PR TITLE
Octowrapper

### LINKED ISSUE #18: Make a Wrapper class in TypeScript to make it more convenient to access stuff from Octokit
I think we should have a wrapper class to make it easier to access functions from Octokit.

### PR DESCRIPTION
**Pull request by ilumsden**
Closes #18.

Implements a wrapper for the Octokit API that could theoretically be used for GitLab too. Also adds classes for data that would be output from the API so that we don't have to deal with `any` or Octokit's ridiculous types.

### CODE REVIEW

File: lib/git-interface/comment.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+export class GitComment
+{
+    public owner: string;
+    public repo: string;
+    public issue_id: number;
+    public id: string;
+    public body: string;
```
Comment (robobenklein) [inline lib/git-interface/comment.ts]: we should be able to get both the rawtext and the html versions of the comment via githubs api, do we have access to those here?

File: lib/git-interface/comment.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+export class GitComment
+{
+    public owner: string;
+    public repo: string;
+    public issue_id: number;
```
Comment (robobenklein) [inline lib/git-interface/comment.ts]: comments could also be on PRs and MRs, right? for github those IDs are unique, but maybe not for gitlab

File: lib/git-interface/pull_request.d.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { Issue } from "./issue"
+
+export interface PullRequest extends Issue
```
Comment (robobenklein) [inline lib/git-interface/pull_request.d.ts]: future-todo: gitlab has `Merge Requests` instead of PRs

**Comment by robobenklein** [comment]
What are the chances we can get compilation testing up on this?

I'd be all for moving this to Sprint 3 and putting something like travis-ci on the job of running `npm run compile`

File: lib/git-interface/comment.ts (hunk 2 above)
Comment (ilumsden) [reply lib/git-interface/comment.ts]: I just checked. GitLab's comments (called notes in the API) at least appear to be unique.

File: lib/git-interface/comment.ts (hunk 1 above)
Comment (ilumsden) [reply lib/git-interface/comment.ts]: Not currently. I'm looking into how to do that.

File: lib/git-interface/pull_request.d.ts (hunk 3 above)
Comment (ilumsden) [reply lib/git-interface/pull_request.d.ts]: I don't know if this is super necessary. At the end of the day, `Issue` and `PullRequest` are interfaces. We'd have separate implementations for GitHub and GitLab. The naming difference could be resolved there.

File: lib/git-interface/comment.ts (hunk 1 above)
Comment (ilumsden) [reply lib/git-interface/comment.ts]: Various URLs (both API and HTML) are now stored in each of the classes and interfaces.

**Comment by ilumsden** [comment]
@robobenklein, I added a basic `.travis.yml` file to the branch for compilation testing. Since `npm run compile` doesn't seem to work, I used `npm run test` and `npm run build` as the scripts.

**Comment by ilumsden** [comment]
@robobenklein, @atltt my code should be ready for "final" review. I'd appreciate it if you guys could take a look and let me know if there's anything else I should add or change.

**Comment by robobenklein** [comment]
`++ npm install` to the travis.yml and see if it works?

File: lib/git-interface/comment.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        this.updated_at = updated_at;
+    }
+}
+
+export class GitPRComment extends GitComment
+{
+    public path: string; // path to file that necessitates the comment
+    public diff_position: number;
+
+    constructor(c_body: string, id=0, comment_id="", url="", html_url="", user=new User(),
```
Comment (robobenklein) [inline lib/git-interface/comment.ts]: if you want, `prettier.ts` is working and you can autoformat with `npm run prettier`

File: lib/tsconfig.json [hunk 5]
```diff
@@ -1,5 +1,9 @@
 {
   "compilerOptions": {
+    "baseUrl": "..",
```
Comment (robobenklein) [inline lib/tsconfig.json]: does this mean we will need to change the `outDir`/source or is that still fine?

**Comment by robobenklein** [comment]
!!! 
image

**Pull request merged**